### PR TITLE
Supports `--release` option for `javadoc` task

### PR DESCRIPTION
**Pull request by scaventz**
Signed-off-by: Xin Wang <scaventz@outlook.com>

Fixes [#18274](https://github.com/gradle/gradle/issues/18274)

### Contributor Checklist
- [x] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [x] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [x] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g. the code was written by yourself or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [x] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [x] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [x] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [ ] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [x] Ensure that tests pass sanity check: `./gradlew sanityCheck`
- [x] Ensure that tests pass locally: `./gradlew <changed-subproject>:quickTest`

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

### CODE REVIEW

File: subprojects/language-java/src/main/java/org/gradle/external/javadoc/CoreJavadocOptions.java [hunk 1]
```diff
@@ -538,6 +541,22 @@ public MinimalJavadocOptions sourceNames(String... sourceNames) {
...
 
+    @Override
+    public Integer getRelease() {
+        return release.getValue() == null ? null : Integer.valueOf(release.getValue());
+    }
+
+    @Override
+    public void setRelease(Integer release) {
+        this.release.setValue(release == null ? null : release.toString());
+    }
```
Comment (scaventz) [inline subprojects/language-java/src/main/java/org/gradle/external/javadoc/CoreJavadocOptions.java:552]: Same `Integer` type as `release` in `JavaCompile`.

**Comment by ljacomet** [comment]
@asodja Since you re-opened this, do you plan on following up? 
If so, could you assign it to yourself?

**Comment by asodja** [comment]
Assigned to me. I think if this is a valid Javadoc option and PR is here, it's worth spending some time on it.

**Comment by Scrappers-glitch** [comment]
> I think if this is a valid Javadoc option and PR is here, it's worth spending some time on it.

Yes it's a valid javadoc option: 
```bash
┌─[pavl-machine@pavl-machine]─[~]
└──╼ $javadoc --help | grep release
                  releases
                  either -source or --release.
    --release <release>
                  Provide source compatibility with specified release
    --source <release>, -source <release>
                  Provide source compatibility with specified release
                  Override location of system modules used for modular releases
    --since <release>(,<release>)*
                  Document new and deprecated API in the specified releases
```
And it's missing from the currently defined core javadoc options at [CoreJavadocOptions.java](https://github.com/gradle/gradle/blob/master/subprojects/language-java/src/main/java/org/gradle/external/javadoc/CoreJavadocOptions.java).

@asodja If you are busy with other issues, you may assign me to this issue as a starting tour for a new fresh contributor :-).

**Pull request closed without merging**